helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -746,13 +705,11 @@ public void deleteAndWaitForCompletion(String workflow, long timeout)
     BaseDataAccessor baseDataAccessor = _accessor.getBaseDataAccessor();
     PropertyKey.Builder keyBuilder = _accessor.keyBuilder();
 
-    String idealStatePath = keyBuilder.idealStates(workflow).getPath();
     String workflowConfigPath = keyBuilder.resourceConfig(workflow).getPath();
     String workflowContextPath = keyBuilder.workflowContext(workflow).getPath();
 
     while (System.currentTimeMillis() <= endTime) {
-      if (baseDataAccessor.exists(idealStatePath, AccessOption.PERSISTENT)
-          || baseDataAccessor.exists(workflowConfigPath, AccessOption.PERSISTENT)
+      if (baseDataAccessor.exists(workflowConfigPath, AccessOption.PERSISTENT)
```

Comment:
Can we keep TaskDriver to clean up IS for a while? What would happen to the workflows/jobs that are running if we switch to new controller? Are these IS are getting deleted anyways? What do you think?

Reply:
So this section you commented on is only used for testing purpose, as far as I know. 

However, you raised a good point here: 
1. Workflows relying on the current pipeline logic should be fine during the switch, because they are guaranteed to have workflow configs.
2. Workflows relying on the old pipeline logic (no config) wouldn't be managed by the controller after this change. Is that a concern?

In general, keeping the IS clean up has only one advantage, which is to clean up the leftover IS from pre-existing workflows. However I don't think that is an huge issue.

Reply:
@alirezazamani Could you take another look at this?